pkg/kube/watcher.go

```diff
@@ -138,76 +148,76 @@ func WatchForServices(
 				continue
 			case event := <-eventCh:
 				if event.deleted {
-					if enablePrivilegedService {
-						if err := portTracker.Remove(string(event.UID)); err != nil {
-							log.Errorw("failed to delete a port from tracker", log.Fields{
-								"error":     err,
-								"UID":       event.UID,
-								"ports":     event.portMapping,
-								"namespace": event.namespace,
-								"name":      event.name,
-							})
-						} else {
-							log.Debugf("kubernetes service: port mapping deleted %s/%s:%v",
-								event.namespace, event.name, event.portMapping)
+					if enableListeners {
+						for port := range event.portMapping {
+							if err := portTracker.RemoveListener(ctx, k8sServiceListenerIP, int(port)); err != nil {
+								log.Errorw("failed to close listener", log.Fields{
```

Comment:
Shouldn't the fields include `port`? Otherwise we can't tell _which_ port failed to close…